crates/load-cargo/src/lib.rs

```diff
@@ -44,7 +44,13 @@ pub fn load_workspace_at(
     load_config: &LoadCargoConfig,
     progress: &dyn Fn(String),
 ) -> anyhow::Result<(RootDatabase, vfs::Vfs, Option<ProcMacroClient>)> {
-    let root = AbsPathBuf::assert_utf8(std::env::current_dir()?.join(root));
+    // Use the provided path as is if it is absolute.
+    // Otherwise, join it with the current dir path.
+    let root = AbsPathBuf::assert_utf8(if root.is_absolute() {
+        root.to_path_buf()
+    } else {
+        std::env::current_dir()?.join(root)
+    });
```

Comment:
This seems unnecessary? 
Quoting the docs of `Path::join`

> If `path` is absolute, it replaces the current path.

`path` being the argument to `join` here, that is in our case `root`. So this change shouldn't do anything here

Reply:
Ah, right. In that case, something else locally was breaking the workspace discovery. Apologies for the noise.